src/makeInfinityAdder.js

```diff
@@ -4,7 +4,20 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(num) {
+    if (num === undefined) {
+      const finalSum = sum;
+
+      sum = 0;
+
+      return finalSum;
+    }
+    sum += num;
+
+    return adder;
```

Comment:
The `adder` function should handle cases where `num` is not a number. Currently, if a non-numeric value is passed, it will be added to `sum`, which can lead to unexpected results. Consider adding a check to ensure `num` is a number before adding it to `sum`.